Kernel/FileSystem/BlockBasedFileSystem.cpp

```diff
@@ -114,6 +114,25 @@ BlockBasedFileSystem::~BlockBasedFileSystem()
 {
 }
 
+bool BlockBasedFileSystem::initialize()
+{
+    VERIFY(block_size() != 0);
+    auto cached_block_data = KBuffer::try_create_with_size(DiskCache::EntryCount * block_size());
+    if (!cached_block_data)
+        return false;
+
+    auto entries_data = KBuffer::try_create_with_size(DiskCache::EntryCount * sizeof(CacheEntry));
+    if (!entries_data)
+        return false;
+
+    auto disk_cache = adopt_own_if_nonnull(new (nothrow) DiskCache(const_cast<BlockBasedFileSystem&>(*this), cached_block_data.release_nonnull(), entries_data.release_nonnull()));
```

Comment:
`const_cast` when the method isn't `const`?

Reply:
Yup, was just a blind copy when I moved the code. Thanks for catching.